src/Components/Generator/DynamicInputContainer.js

```diff
@@ -216,13 +216,34 @@ export default class DynamicInputContainer extends preact.Component {
     }
 
     removeDynamicInput(event) {
-        if (window.confirm(t('confirm-input-remove', 'generator'))) {
-            this.setState(prev => {
-                delete prev.fields[event.target.getAttribute('rel')];
-                return prev;
-            });
-            this.pushStateUp();
+        // prompt only if respective field has value entered while removing
+        if (!this.checkInputData(event)) {
+            this.confirmStateChange(event);
+        } else if(window.confirm(t('confirm-input-remove', 'generator'))) {
+            this.confirmStateChange(event);
+        }   
+    }
+
+    confirmStateChange(event) {
+        this.setState(prev => {
+            delete prev.fields[event.target.getAttribute('rel')];
+            return prev;
+        });
+        this.pushStateUp();
+    }
+
+    // returns whether the field has value based on its type
+    checkInputData(event) {
+        let val = event.target.getAttribute('rel');
+        let name = this.state.fields[val].value;
```

Comment:
These can be `const` but I guess the Prettier will take care of that. But the variable names are also misleading here. Just from reading the names I would assume `val` to contain the field value and `name` to contain its name. Maybe call them something like `field` and `field_value`. 

Also, to make this method more generic you could move the event to `removeDynamicInput()` and just pass down the field id.